### PR TITLE
Space Traveler - Complete Project

### PR DESCRIPTION
**Pull request by Donmark2k**
### Highlights 🥇

**Space travelers**

**Learning objectives**

- Use React documentation
- Use React components
- Use React props
- Use React Router
- Connect React and Redux
- Handle events in a React app
- Write unit tests with React Testing Library
- Use styles in a React app
- Use React hooks
- Apply React best practices and language style guides in code
- Use store, actions, and reducers in React
- Perform a code review for a team member

**What We Have Done:**

- Added liters. The [react & redux linter](https://github.com/microverseinc/linters-config/tree/master/react-redux).
- We create a git flow.
- We deleted all the boilerplate from Create React App (text, images, styles).
-  We added a reducer
-  We created a store 
-  We connect to API
-  We style the rocket page
- We added the test functionality

**Followed Rules:**

- Made sure [no linter errors](https://github.com/microverseinc/linters-config)
- Followed [Git Flow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md)
- Documented [professionaly](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md)
- Followed the list of [best practices for react & redux](https://github.com/microverseinc/linters-config/tree/master/react-redux).

### CODE REVIEW

File: src/App.test.js [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+  });
+  it('Rockets fetch data from API', async () => {
+    const url = 'https://api.spacexdata.com/v4/rockets';
+    const axiosSpy = jest.spyOn(axios, 'get');
+    jest.setTimeout(90000);
+    const dispatchSpy = jest.fn();
+    await fetchRockets(url)(dispatchSpy);
+    expect(axiosSpy).toHaveBeenCalledWith(url);
+  });
+});
```
Comment (muneebulrehman) [inline src/App.test.js:115]: - Optional
I would recommend putting separate tests in separate files. Eg., the tests for testing Rocket reducers should be in `RocketReducer.test.js`.

File: src/redux/missions/MissionsSlice.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        if (mission.mission_id === payload.payload) {
+          if (mission.reserved === true) {
+            return { ...mission, reserved: false };
+          }
+          return { ...mission, reserved: true };
+        }
+        return mission;
+      }),
+    }),
+  },
```
Comment (muneebulrehman) [inline src/redux/missions/MissionsSlice.js:35]: - Optional
I strongly recommend having a separate reducer for leaving the mission. It is always recommended to have one reducer per action.

File: src/redux/rockets/RocketsSlice.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        if (rocket.id === payload.payload) {
+          if (rocket.reserved === true) {
+            return { ...rocket, reserved: false };
+          }
+          return { ...rocket, reserved: true };
+        }
+        return rocket;
+      }),
+    }),
+  },
```
Comment (muneebulrehman) [inline src/redux/rockets/RocketsSlice.js:37]: - Optional
I strongly recommend having a separate reducer for canceling the rocket reservation. It is always recommended to have one reducer per action.

**Review by muneebulrehman — APPROVED** [review]
Hi team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it 🥳
Congratulations! 🎉 

# Status: Approved✔️

## Highlights
- No linter errors
- Descriptive PR
- All technical requirements met for this milestone.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/App.test.js (hunk 1 above)
Comment (zewdu444) [reply src/App.test.js:115]: @muneebulrehman thank you for the feedback we will improve accordingly

File: src/redux/rockets/RocketsSlice.js (hunk 3 above)
Comment (zewdu444) [reply src/redux/rockets/RocketsSlice.js:37]: @muneebulrehman thank you for the feedback we will improve accordingly

**Pull request merged**